transport/uml/clearConversation/clear_conversation_hld.puml

```diff
@@ -0,0 +1,54 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+!theme bluegray
+!pragma teoz true
+
+skinparam backgroundColor #whitesmoke
+title End user clear
+
+participant App
+box TransportSDK #lightcyan
+participant MessagingClient
+end box
+box BackEnd
+participant Shyrka
+participant Squonk
+participant GenesysCloud
+end box
+participant Agent
+
+rnote over MessagingClient
+  Given:
+  * Session is connected.
+  * User wants to delete conversation.
+end rnote
+
+App -> MessagingClient: clearConversation()
+MessagingClient -> Shyrka: send PresenceClearEvent
+Shyrka -> Squonk: Check Config and conversationClear option
+Squonk --> Shyrka: retrieve config
+alt #aaff8833 deployment has configurationClear enabled
+   Shyrka --> MessagingClient: 200 response on PresenceClearEvent
+   MessagingClient -> MessagingClient: ignore response
+   Shyrka -> MessagingClient: SessionClearedEvent
+   MessagingClient -> MessagingClient: invalidateConversationCache()
+   MessagingClient -> App: eventListener.onEvent(Event.Clear)
+rnote over App
+    Application is expected to remove messages from UI.
+end rnote
+   Shyrka -> Shyrka: Clear all session information
+   Shyrka -> MessagingClient: ConnectionClosedEvent
+   MessagingClient -> MessagingClient: disconnect()
+   MessagingClient -> App: onStateChanged(Closed)
+   MessagingClient -> App: eventListener.onEvent(Event.ConnectionClosed)
+   Shyrka -> GenesysCloud: Send message (presence clear)
+   GenesysCloud -> GenesysCloud: Produce end user clear event
+   GenesysCloud -> Agent: Notify agent of clear
+   Agent --> GenesysCloud: Supply Wrapup
+else #ffaaff33 deployment does NOT have configurationClear enabled or any other failure
```

Comment:
1. typo: `configurationClear` should be `conversationClear`
2. In case the conversation clear is disabled we won't have the error passing section, right?

Reply:
1. Will fix the typo.
2. In case of conversation clear is disabled I will get: {"type":"response","class":"string","code":403,"body":"Presence events Conversation Clear are not supported"}
Which will be mapped by transport to 
``` 
Event.Error(
    code = ErrorCode.ClientRequestError(403), 
    message = "Presence events Conversation Clear are not supported"
    correctiveAction = CorrectiveAction.Forbidden
```
I am going to add the deploymentConfig.isClearEnabled to diagram before sending the request. Good catch!
3. Hope the answer in section `2` answer this question as well.